net/gateway/connect.go

```diff
@@ -0,0 +1,217 @@
+package gateway
+
+import (
+	"crypto/ed25519"
+	"errors"
+	"fmt"
+	"go.sia.tech/core/net/mux"
+	"go.sia.tech/core/types"
+	"io"
+	"net"
+)
+
+var (
+	errRejectedVersion = errors.New("peer rejected our version")
+)
+
+const (
+	currentVersion uint8 = 1
+	minimumVersion uint8 = 1
+	rejectResponse       = "reject"
+	acceptResponse       = "accept"
+	// listener / tcp connections not implemented yet
+	dummyAddr = "127.0.0.1:1"
+)
+
+type SessionConfig struct {
+	publicKey  ed25519.PublicKey
+	genesisID  types.BlockID
+	uniqueID   GatewayID
+	netAddress string
```

Comment:
these fields will need to be exported as well.
`GatewayID` can just be `[8]byte`, I think a separate type is overkill.